hudi-common/src/main/java/org/apache/hudi/common/util/collection/DiskBasedMap.java

```diff
@@ -111,8 +111,14 @@ private void initFile(File writeOnlyFile) throws IOException {
     if (writeOnlyFile.exists()) {
       writeOnlyFile.delete();
     }
-    if (!writeOnlyFile.getParentFile().exists()) {
-      writeOnlyFile.getParentFile().mkdir();
+    File parentDir = writeOnlyFile.getParentFile();
+    if (!parentDir.exists()) {
+      parentDir.mkdir();
+      // we should set all right for other user.
```

Comment:
`we should set all rights for all users, otherwise some users cannot access this directory and cause query to fail?` and is it possible to add unit test for this?

Reply:
ok， i will try to add UT。 thanks

Reply:
@leesf  sorry, i think it's hard to simulate different users, since we have no right to create a new user in linux（authority issues）. hope you can give me some advice